tsnet/tsnet.go

```diff
@@ -254,11 +265,7 @@ func (s *Server) start() (reterr error) {
 		Stderr:     io.Discard, // log everything to Buffer
 		Buffer:     s.logbuffer,
 		NewZstdEncoder: func() logtail.Encoder {
-			w, err := smallzstd.NewEncoder(nil)
```

Comment:
The `logtail.Encoder` interface has a `Close` method.

So this change is risky... if we later correctly change logtail to fix the `Close` leak, then this code breaks when the first tsnet server shuts down, causing the other still-running Servers to have broken logtail clients.

So two options:

1) wrap the returned zstdEncoder in a type with a dummy no-op Close method,
2) just fix the missing Close in logtail, making this PR no longer required

I'm curious why (2) isn't easier.

Reply:
I don't actually see any obviously missing Close/Shutdown call in tsnet or logtail.

Is tsnet.Server.Close being called?

Reply:
> Is tsnet.Server.Close being called?

I think so:
https://github.com/tailscale/corp/blob/373f00bebd69d022c44145c95c65700c74518fb2/cmd/nodecop/nodecop.go#L118

The resource being leaked is allocated at https://github.com/klauspost/compress/blob/03c136c2259e3adcfae5c4a050ce6de1f3330597/zstd/encoder.go#L87, and I wonder if storing it in a channel on the next line is confusing the GC.

Reply:
The GC is only relevant in that it can prevent a finalizer from running. But if you're relying on a finalizer for correctness, you've already kinda lost.

So I'd worry more about where we're not explicitly closing things.

Reply:
The `tsnet.Server` was not calling `Close` on the `netstack.Impl` it created in `Start`. This was fixed in #5809.